lib/node/fullnode.js

```diff
@@ -603,6 +627,19 @@ class FullNode extends Node {
 
     return false;
   }
+
+  /**
+   * Retrieve compact filter by hash.
+   * @param {Hash} hash
+   * @returns {Promise} - Returns {@link Buffer}.
+   */
+
+  async getBlockFilter(hash) {
+    if (this.filterindex)
```

Comment:
This won't work with height as a param, as used in `lib/node/http.js`:

```patch
   /**
    * Retrieve compact filter by hash.
-   * @param {Hash} hash
+   * @param {Hash|Number} hash
    * @returns {Promise} - Returns {@link Buffer}.
    */
 
   async getBlockFilter(hash) {
-    if (this.filterindex)
-      return this.filterindex.getFilter(hash);
+    if (!this.filterindex)
+      return null;
 
-    return null;
+    if (typeof hash === 'number')
+      hash = await this.chain.getHash(hash);
+
+    if (!hash)
+      return null;
+
+    return this.filterindex.getFilter(hash);
   }
 }
```

Reply:
Fixed.